test/test_main.py

```diff
@@ -1128,10 +1128,22 @@ def _call_event(self, rel_path: Path, env: typing.Dict[str, str]):
         )
         subprocess.run(
             [sys.executable, str(dispatch)],
-            # stdout=self.stdout,
-            # stderr=self.stderr,
+            stdout=self.stdout,
+            stderr=self.stderr,
             check=True, env=env, cwd=str(self.JUJU_CHARM_DIR))
 
+    def test_crash_action(self):
+        self._prepare_actions()
+        self.stderr = tempfile.TemporaryFile()
+        self.addCleanup(self.stderr.close)
+        fake_script(typing.cast(unittest.TestCase, self), 'action-get', "echo '{}'")
+        with self.assertRaises(subprocess.CalledProcessError):
+            self._simulate_event(EventSpec(
+                ops.ActionEvent, 'keyerror_action',
+                env_var='JUJU_ACTION_NAME'))
+        self.stderr.seek(0)
+        self.assertIn(b'KeyError', self.stderr.read())
```

Comment:
I also think we should test that the exception value (string) is included in the output (just not the "Uncaught" and "Use juju debug-log" bits). Perhaps:

```suggestion
        stderr = self.stderr.read()
        self.assertIn(b'KeyError', stderr)
        self.assertIn(b"'foo' not found in 'bar'", stderr)
```